### PR TITLE
update read_csv and to_csv

### PR DESCRIPTION
**Pull request by muyixi315**
read_csv support StringIO
to_csv support return buffer

### CODE REVIEW

File: py-polars/polars/frame.py [hunk 1]
```diff
@@ -41,6 +43,15 @@
     from .lazy import LazyFrame, Expr
 
 
+def to_csv(df, path_or_buf=None):
```
Comment (ritchie46) [inline py-polars/polars/frame.py:46]: Could you put this logic in `DataFrame.to_csv`? I don't want `to_csv` as a different function as that would not be in line with the rest of the API.

**Review by ritchie46 — COMMENTED** [review]
I don't really understand the purpose of this pull request? You already write to a buffer:

```python
import polars as pl
import io

df = pl.DataFrame({"foo": [1, 2], "bar": [None, 1]})

f = io.BytesIO()
df.to_csv(f)
f.seek(0)

print(f.read())
```

```
> b'foo,bar\n1,\n2,1\n'
```

**Comment by muyixi315** [comment]
> I don't really understand the purpose of this pull request? You already write to a buffer:
> 
> ```python
> import polars as pl
> import io
> 
> df = pl.DataFrame({"foo": [1, 2], "bar": [None, 1]})
> 
> f = io.BytesIO()
> df.to_csv(f)
> f.seek(0)
> 
> print(f.read())
> ```
> 
> ```
> > b'foo,bar\n1,\n2,1\n'
> ```

mainly consider the usage of to_csv compatible with pandas，so ...

File: py-polars/polars/frame.py (hunk 1 above)
Comment (ritchie46) [inline py-polars/polars/frame.py:46]: Could you put the logic in `DataFrame.to_csv` and remove this function? I want the API similar to the rest of IO.

File: py-polars/polars/functions.py [hunk 2]
```diff
@@ -14,15 +15,44 @@
...
+_VALID_URLS = set(uses_relative + uses_netloc + uses_params)
+_VALID_URLS.discard("")
 
 
 def _process_http_file(path: str) -> io.BytesIO:
     with urllib.request.urlopen(path) as f:
         return io.BytesIO(f.read())
 
 
+def _is_url(url):
```
Comment (ritchie46) [inline py-polars/polars/functions.py:36]: Could you add the types in the function description encapsulated by quotation marks (this is needed for python 3.6 support)

**Review by ritchie46 — CHANGES_REQUESTED** [review]
> mainly consider the usage of to_csv compatible with pandas，so ...

Ok, I think it's ok to return a `StringIO` when not file is passed to `DataFrame.to_csv`. But I want to keep the API lean and only have the `to_csv` method on the `DataFrame` object and not also as a separate function, could you make those changes?

The CI pipeline is also failing. [Here is the contribution guide](https://github.com/ritchie46/polars/blob/master/CONTRIBUTING.md) for the linting and formatting steps.

File: py-polars/polars/frame.py (hunk 1 above)
Comment (muyixi315) [reply py-polars/polars/frame.py]: Modified, please review. Thank you for your guidance.

File: py-polars/polars/functions.py (hunk 2 above)
Comment (muyixi315) [reply py-polars/polars/functions.py]: Modified, please review. Thank you for your guidance.

File: py-polars/polars/frame.py (hunk 1 above)
Comment (muyixi315) [reply py-polars/polars/frame.py]: Modified, please review. Thank you for your guidance.

File: py-polars/polars/frame.py [hunk 3]
```diff
@@ -305,7 +307,7 @@ def to_pandas(
 
     def to_csv(
         self,
-        file: Union[TextIO, str, Path],
+        file: Union[TextIO, str, Path] = None,
```
Comment (ritchie46) [inline py-polars/polars/frame.py:310]: Now that we also accept a `None`, this should be `Optional[Union[TextIO, str, Path]]`

**Comment by ritchie46** [comment]
LGTM overall. Only one small remark on the type of `DataFrame.to_csv`

File: py-polars/polars/frame.py (hunk 3 above)
Comment (muyixi315) [reply py-polars/polars/frame.py]: Ok, I get it now, thanks！

**Comment by ritchie46** [comment]
Thanks for your contrinution @muyixi315 , Looks good now. I will merge this in, once the issues with CI are solved.

**Comment by muyixi315** [comment]
Ok, This python code format problem with CI has been resolved. I installed `black==21.4b2` and ran `$ black`.

**Comment by muyixi315** [comment]
> Thanks for your contrinution @muyixi315 , Looks good now. I will merge this in, once the issues with CI are solved.

Ok, This python code format problem with CI has been solved. I installed `black==21.4b2` and ran `$ black`.

**Pull request merged**